lib/ansible/modules/cloud/amazon/ecs_tag.py

```diff
@@ -0,0 +1,238 @@
+    type: str
+  tags:
+    description:
+      - A dictionary of tags to add or remove from the resource.
+      - If the value provided for a tag is null and C(state) is I(absent), the tag will be removed regardless of its current value.
+    type: dict
+  purge_tags:
+    description:
+      - Whether unspecified tags should be removed from the resource.
+      - "Note that when combined with C(state: absent), specified tags with non-matching values are not purged."
+    type: bool
+    default: no
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+EXAMPLES = r'''
+- name: Ensure tags are present on a resource
+  ecs_tag:
+    cluster_name: mycluster
+    resource_type: cluster
+    state: present
+    tags:
+      Name: ubervol
+      env: prod
+
+- name: Retrieve all tags on a cluster
+  ecs_tag:
+    cluster_name: mycluster
+    resource: http_task
+    resource_type: task
+    state: list
+
+- name: Remove the Env tag
+  ecs_tag:
+    cluster_name: mycluster
+    resource_type: cluster
+    tags:
+      Env:
+    state: absent
+
+- name: Remove the Env tag if it's currently 'development'
+  ecs_tag:
+    cluster_name: mycluster
+    resource_type: cluster
+    tags:
+      Env: development
+    state: absent
+
+- name: Remove all tags except for Name from a cluster
+  ecs_tag:
+    cluster_name: mycluster
+    resource_type: cluster:27
+    tags:
+        Name: foo
+    state: absent
+    purge_tags: true
+'''
+
+RETURN = r'''
+tags:
+  description: A dict containing the tags on the resource
+  returned: always
+  type: dict
+added_tags:
+  description: A dict of tags that were added to the resource
+  returned: If tags were added
+  type: dict
+removed_tags:
+  description: A dict of tags that were removed from the resource
+  returned: If tags were removed
+  type: dict
+'''
+
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import boto3_tag_list_to_ansible_dict, ansible_dict_to_boto3_tag_list, compare_aws_tags
+
+try:
+    from botocore.exceptions import BotoCoreError, ClientError
+except Exception:
+    pass    # Handled by AnsibleAWSModule
+__metaclass__ = type
+
+
+def get_tags(ecs, module, resource):
+    try:
+        return boto3_tag_list_to_ansible_dict(ecs.list_tags_for_resource(resourceArn=resource)['tags'])
+    except (BotoCoreError, ClientError) as e:
+        module.fail_json_aws(e, msg='Failed to fetch tags for resource {0}'.format(resource))
+
+
+def main():
+    argument_spec = dict(
+        cluster_name=dict(required=True),
+        resource=dict(required=False),
+        tags=dict(type='dict'),
+        purge_tags=dict(type='bool', default=False),
+        state=dict(default='present', choices=['present', 'absent', 'list']),
+        resource_type=dict(default='cluster', choices=['cluster', 'task', 'service', 'task_definition', 'container'])
+
+    )
+    required_if = [('state', 'present', ['tags']), ('state', 'absent', ['tags'])]
```

Comment:
I think the code below is going to spit out an ugly exception if resource_type is task/service/task_definition/container and resource is undefined.

It's probably better to require resource be set if resource_type is one of those values...

```suggestion
    required_if = [
        ('state', 'present', ['tags']),
        ('state', 'absent', ['tags']),
        ('resource_type', 'task', ['resource']),
        ('resource_type', 'service', ['resource']),
        ('resource_type', 'task_definition', ['resource']),
        ('resource_type', 'container', ['resource']),
    ]
```

(note: I've not double checked the spacing here, so it'll possibly generate a lint warning)